### PR TITLE
Merge node-addon-api

### PR DESCRIPTION
**Pull request by almilosz**
### Details:
 - Merge changes with js-wasm-proto
 - Move ./assets to the parent folder, so both wasm api  and node-addon api can use them
 -  Add hello_world javascript jupyter notebook
 - Add methods required to implement ISession, IShape, IModel interfaces
 - Refactor of source files
 - Add tensor JEST tests

### Tickets:
 - 102916
 - 102902

### CODE REVIEW

**Review by vishniakov-nikolai — CHANGES_REQUESTED** [review]
Please, check files with which have empty last line warning.
I left several minor comments. After them fix it is possible to merge.

File: samples/js/assets/images/coco.jpg
[the remark this answers is not in the document]
Comment (almilosz) [reply samples/js/assets/images/coco.jpg:1]: I will remove this photo and add a different one that has its original shape and needs resizing.

File: samples/js/node-addon/README.md
[the remark this answers is not in the document]
Comment (almilosz) [reply samples/js/node-addon/README.md:1]: Done

File: samples/js/node-addon/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+## Quick start
+You should install OpenVINO™ Runtime on your system first.
+[Install OpenVINO™ Runtime on Linux from an Archive File guide](https://docs.openvino.ai/latest/openvino_docs_install_guides_installing_openvino_from_archive_linux.html#doxid-openvino-docs-install-guides-installing-openvino-from-archive-linux)
+
+You should also move here native addon module `Release/ov_node_addon.node` from the build directory.
+
+To run the example you need to pass a path to an image and a path to the required model.
+```sh
+$ npm install
+$ node <example-name> <path-to-image> <path-to-model>
```
[the remark this answers is not in the document]
Comment (almilosz) [reply samples/js/node-addon/README.md:12]: I am not sure what you meant, but I changed it and now it is optional to pass image path. We can get back to it later

File: samples/js/node-addon/imagenet_2012_labels.txt
[the remark this answers is not in the document]
Comment (almilosz) [reply samples/js/node-addon/imagenet_2012_labels.txt:1]: It is possible. I was just wondering why is it better to create a .mjs file instead using simple .txt and reading it

File: src/bindings/js/node-addon/package.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{
+    "name": "hello",
```
[the remark this answers is not in the document]
Comment (almilosz) [reply src/bindings/js/node-addon/package.json:2]: I renamed it to "openvinojs-node"

File: samples/js/node-addon/package.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    "private": true,
+
+    "dependencies": {
+      "bindings": "~1.2.1",
+      "jimp": "^0.16.1",
+      "opencv.js": "^1.2.1"
+    },
+
+    "scripts": {
+        "preinstall": "echo ${INTEL_OPENVINO_DIR?\\$INTEL_OPENVINO_DIR should set before installation. Please execute 'source ~/intel/openvino/setupvars.sh' and run 'npm i' again.}"
```
[the remark this answers is not in the document]
Comment (almilosz) [reply samples/js/node-addon/package.json:13]: Path to INTEL_OPENVINO_DIR  is still required. Current node addon binding does not contain C++ binaries.

File: samples/js/assets/images/coco.jpg
[the remark this answers is not in the document]
Comment (vishniakov-nikolai) [reply samples/js/assets/images/coco.jpg:1]: So cute! 😻

**Pull request merged**